rbx_binary/src/deserializer.rs

```diff
@@ -674,7 +674,36 @@ impl<R: Read> BinaryDeserializer<R> {
                     });
                 }
             },
-            Type::BrickColor => {}
+            Type::BrickColor => match canonical_type {
+                VariantType::BrickColor => {
+                    let mut values = vec![0; type_info.referents.len()];
+                    chunk.read_interleaved_u32_array(&mut values)?;
+
+                    for (value, referent) in values.into_iter().zip(&type_info.referents) {
+                        let instance = self.instances_by_ref.get_mut(referent).unwrap();
+                        let color = value
+                            .try_into()
+                            .map(|value| BrickColor::from_number(value).unwrap())
```

Comment:
We should not use `unwrap` in the main serializer/deserializer codebase: we don't want to panic if a file has an invalid BrickColor value: we should instead return an error.

I would write this chain with something like:

```rust
let color = value
    .try_into()
    .ok() // turn `Result<T, E>` into `Option<T>`, since we don't use the error anyways
    .and_then(BrickColor::from_number)
    .ok_or_else(|| InnerError::InvalidPropData {
        type_name: type_info.type_name.clone(),
        prop_name: prop_name.clone(),
        valid_value: "a valid BrickColor",
        actual_value: value.to_string(),
    })?;
```

Reply:
Ah darn it, I knew I forgot something. That's on me, I know better than to leave `unwrap` in production code.

I also forgot that `ok` was a thing. That's a lot more elegant. I'll update it tomorrow (or if you want to suggest a change, I'll accept it).

Reply:
Let's just wait until tomorrow!